components/brave_shields/content/browser/ad_block_custom_filters_provider.cc

```diff
@@ -5,14 +5,41 @@
 
 #include "brave/components/brave_shields/content/browser/ad_block_custom_filters_provider.h"
 
+#include <optional>
+#include <string>
 #include <utility>
 #include <vector>
 
+#include "base/strings/strcat.h"
+#include "base/strings/string_tokenizer.h"
 #include "base/task/single_thread_task_runner.h"
 #include "brave/components/brave_shields/core/browser/ad_block_filters_provider_manager.h"
 #include "brave/components/brave_shields/core/common/pref_names.h"
 #include "components/prefs/pref_service.h"
 
+namespace {
+
+std::optional<std::string> RemoveFiltersForHost(
+    const std::string& host,
+    const std::string& custom_filters) {
+  if (custom_filters.empty()) {
+    return std::nullopt;
+  }
+
+  std::stringstream result;
```

Comment:
I don't see any good reason to use streams in this place, but there is no explicit rule in the code style that forbids it.